test/functional/bundleadd/add-json.bats

```diff
@@ -0,0 +1,84 @@
+#!/usr/bin/env bats
+
+# Author: Castulo Martinez
+# Email: castulo.martinez@intel.com
+
+load "../testlib"
+
+test_setup() {
+
+	create_test_environment "$TEST_NAME"
+	files="/file1,/file2,/file3,/file4,/file5,/file6,/file7,/file8,/file9,/file10,/file11"
+	create_bundle -n test-bundle -f "$files" "$TEST_NAME"
+
+}
+
+@test "ADD049: Adding a bundle using machine readable output" {
+
+	# the --json-output flag can be used so all the output of the bundle-add
+	# command is created as a JSON stream that can be read and parsed by other
+	# applications interested into knowing real time status of the command
+	# (for example installer so it can provide the user a status of the install process)
+
+	run sudo sh -c "$SWUPD bundle-add --json-output $SWUPD_OPTS test-bundle"
+
+	assert_status_is 0
+	expected_output=$(cat <<-EOM
+		[
+		{ "type" : "start", "section" : "bundle-add" },
```

Comment:
Do we really need that section json? We have only one section for each command and the caller already knows what command we did run.

Reply:
These sections:

```
[
 { "type" : "start", "section" : "bundle-add" },
```
and 
```
{ "type" : "end", "section" : "bundle-add" }
 ]
```
were added for two reasons:
- so we could have a way to instruct the code to open and close an array in the JSON stream "[ .. ]"
- and most importantly so we could remove the "," from the last item in the array. If we just close the array and there is a colon after the last element, the JSON is invalid. There is no easy way to remove that "," from the last element if we remove this cause we don't know when one element is the last.

Reply:
+1 for the "," problem